pkgs/applications/misc/kitty/default.nix

```diff
@@ -33,16 +66,29 @@ buildPythonApplication rec {
       src = ./fix-paths.patch;
       libstartup_notification = "${libstartup_notification}/lib/libstartup-notification-1.so";
     })
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
+    ./no-lto.patch
+    ./no-werror.patch
+    ./png2icns.patch
   ];
 
-  buildPhase = ''
+  buildPhase = if stdenv.isDarwin then ''
+    make app
+  '' else ''
     ${python.interpreter} setup.py linux-package
   '';
 
   installPhase = ''
     runHook preInstall
     mkdir -p $out
+    ${if stdenv.isDarwin then ''
+    mkdir "$out/bin"
+    ln -s ../Applications/kitty.app/Contents/MacOS/kitty "$out/bin/kitty"
```

Comment:
Can you verify the symlink is correct here? Maybe avoid the relative and use $out to prevent confusionz

Reply:
> Can you verify the symlink is correct here?

Yes, I did verify that the `kitty` symlink works:

```console
$ /nix/store/nqyzdmi844za1vqqjh7n2s719ma15y9f-kitty-0.13.3/bin/kitty --version
kitty 0.13.3 created by Kovid Goyal
$ ls -la /nix/store/nqyzdmi844za1vqqjh7n2s719ma15y9f-kitty-0.13.3/bin/
total 16
dr-xr-xr-x  4 root  wheel  136 Dec 31  1969 .
dr-xr-xr-x  6 root  wheel  204 Dec 31  1969 ..
lrwxr-xr-x  1 root  wheel   46 Dec 31  1969 .kitty-wrapped -> ../Applications/kitty.app/Contents/MacOS/kitty
-r-xr-xr-x  1 root  wheel  427 Dec 31  1969 kitty
$ realpath /nix/store/nqyzdmi844za1vqqjh7n2s719ma15y9f-kitty-0.13.3/bin/.kitty-wrapped
/nix/store/nqyzdmi844za1vqqjh7n2s719ma15y9f-kitty-0.13.3/Applications/kitty.app/Contents/MacOS/kitty
```